### PR TITLE
Add a description of `Cargo.lock` conflicts in the Cargo FAQ

### PR DESCRIPTION
**Pull request by QiangHeisenberg**
### What does this PR try to resolve?

Add a message for issue https://github.com/rust-lang/cargo/issues/12079.

For detailed discussion, please refer to:https://github.com/rust-lang/cargo/pull/12092

### CODE REVIEW

File: src/doc/src/faq.md [hunk 1]
```diff
@@ -259,3 +259,49 @@ Some issues we've seen historically which can cause crates to get rebuilt are:
 If after trying to debug your issue, however, you're still running into problems
 then feel free to [open an
 issue](https://github.com/rust-lang/cargo/issues/new)!
+
+### What does "version conflict" mean and how to resolve it?
+
+> failed to select a version for `x` which could resolve this conflict
```
Comment (weihanglo) [inline src/doc/src/faq.md:265]: Missing a newline here

```suggestion
> failed to select a version for `x` which could resolve this conflict

```

File: src/doc/src/faq.md [hunk 2]
```diff
@@ -259,3 +259,49 @@ Some issues we've seen historically which can cause crates to get rebuilt are:
...
+  dependent library, but the version used is restricted, making it impossible to 
+  determine the correct version, it will also cause conflicts. The error message 
+  will prompt: `all possible versions conflict with previously selected packages`. 
+  You may need to modify the version requirements to make them consistent.
+
+- If there are multiple versions of dependencies in the project, when using 
+  [`direct-minimal-versions`], the minimum version requirements cannot be met, 
+  which will cause conflicts. Please check in the dependency graph of the project, 
+  the version requirements for dependencies are inconsistent. Please modify the 
+  version requirements to make them consistent.
```
Comment (weihanglo) [inline src/doc/src/faq.md:289]: This should be better understanding.

```suggestion
  which will cause conflicts. You may need to modify version requirements of your
  direct dependencies to meet the minimum SemVer version accordingly.
```

**Review by weihanglo — COMMENTED** [review]
Thanks, and sorry I've found some minor issues 😅.

File: src/doc/src/faq.md (hunk 2 above)
Comment (QiangHeisenberg) [reply src/doc/src/faq.md]: Welcome to point out more questions, thanks：）

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks!

@bors r+

**Pull request merged**